packages/block-library/src/search/index.php

```diff
@@ -85,10 +86,12 @@ function render_block_core_search( $attributes ) {
 			$aria_label      = sprintf( 'aria-label="%s"', esc_attr( wp_strip_all_tags( $attributes['buttonText'] ) ) );
 			$button_classes .= ' has-icon';
 
-			$button_internal_markup =
-				'<svg id="search-icon" class="search-icon" viewBox="0 0 24 24" width="24" height="24">
+			$search_button_markup =
+				'<svg id="search-icon-%s" class="search-icon" viewBox="0 0 24 24" width="24" height="24">
 					<path d="M13.5 6C10.5 6 8 8.5 8 11.5c0 1.1.3 2.1.9 3l-3.4 3 1 1.1 3.4-2.9c1 .9 2.2 1.4 3.6 1.4 3 0 5.5-2.5 5.5-5.5C19 8.5 16.5 6 13.5 6zm0 9.5c-2.2 0-4-1.8-4-4s1.8-4 4-4 4 1.8 4 4-1.8 4-4 4z"></path>
 				</svg>';
+
+			$button_internal_markup = sprintf( $search_button_markup, $index++ );
```

Comment:
I would probably escape this to be safe. Something like this?

```
id="%s"
```

```
$button_markup = sprintf( $button_markup, esc_attr( 'search-icon' . $index ) );
```

Reply:
`$instance_id` is already set and incremented [above](https://github.com/WordPress/gutenberg/blob/c149dd0b66921733812a33a7c199fb84cb54a6e1/packages/block-library/src/search/index.php#L31)

Could we just repurpose it rather than creating a new static var?

E.g., 

```diff
--- a/packages/block-library/src/search/index.php
+++ b/packages/block-library/src/search/index.php
@@ -27,7 +27,8 @@ function render_block_core_search( $attributes ) {
 		)
 	);
 
-	$input_id         = 'wp-block-search__input-' . ++$instance_id;
+	$uniq_id          = ++$instance_id;
+	$input_id         = "wp-block-search__input-$uniq_id";
 	$classnames       = classnames_for_block_core_search( $attributes );
 	$show_label       = ( ! empty( $attributes['showLabel'] ) ) ? true : false;
 	$use_icon_button  = ( ! empty( $attributes['buttonUseIcon'] ) ) ? true : false;
@@ -85,10 +86,12 @@ function render_block_core_search( $attributes ) {
 			$aria_label      = sprintf( 'aria-label="%s"', esc_attr( wp_strip_all_tags( $attributes['buttonText'] ) ) );
 			$button_classes .= ' has-icon';
 
-			$button_internal_markup =
-				'<svg id="search-icon" class="search-icon" viewBox="0 0 24 24" width="24" height="24">
+			$button_internal_markup = sprintf(
+				'<svg id="search-icon" class="search-icon-%s" viewBox="0 0 24 24" width="24" height="24">
 					<path d="M13.5 6C10.5 6 8 8.5 8 11.5c0 1.1.3 2.1.9 3l-3.4 3 1 1.1 3.4-2.9c1 .9 2.2 1.4 3.6 1.4 3 0 5.5-2.5 5.5-5.5C19 8.5 16.5 6 13.5 6zm0 9.5c-2.2 0-4-1.8-4-4s1.8-4 4-4 4 1.8 4 4-1.8 4-4 4z"></path>
-				</svg>';
+				</svg>',
+				$uniq_id
+			);
 		}
 
 		$button_markup = sprintf(
```

Reply:
We can also use [`wp_unique_id` ](https://developer.wordpress.org/reference/functions/wp_unique_id/) and re-use single ID. This is what we use in other places.

Reply:
I have addressed the review comment and tested locally. It is working. Unique `id`s are getting generated.